blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxEnumIO.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2016 Rackspace.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.codahale.metrics.Timer;
+import com.datastax.driver.core.*;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.google.common.collect.Table.Cell;
+import com.rackspacecloud.blueflood.io.*;
+import com.rackspacecloud.blueflood.io.serializers.metrics.EnumSerDes;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.types.*;
+import com.rackspacecloud.blueflood.utils.Locators;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * This class holds the utility methods to read/write enum metrics
+ * using Datastax driver.
+ */
+public class DatastaxEnumIO implements AsyncWriter, EnumReader {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DatastaxEnumIO.class);
+    private static final EnumSerDes serDes = new EnumSerDes();
+
+    private static final String INSERT_KEY_COLUMN_VALUE_FORMAT = "INSERT INTO %s (key, column1, value) VALUES (?, ?, ?)";
+    private static final String SELECT_FOR_KEY = "SELECT * FROM %s WHERE key = :locator";
+    private static final String SELECT_FOR_KEY_RANGE_FORMAT = "SELECT * FROM %s WHERE key = :locator AND column1 >= :tsStart AND column1 <= :tsEnd";
```

Comment:
If you want, you can use ```QueryBuilder``` and ```bindMarker()``` to create these strings.  E.g., https://gist.github.com/usnavi/b7920c76b9b36d582d43df22a72c349c#file-gistfile1-txt-L45